pkg/reconciler/apis/apiexportendpointslice/apiexportendpointslice_controller.go

```diff
@@ -224,6 +254,32 @@ func (c *controller) enqueueAllAPIExportEndpointSlices(shard interface{}) {
 	}
 }
 
+// enqueuePartition maps a Partition to APIExportEndpointSlices for enqueuing.
+func (c *controller) enqueuePartition(obj interface{}) {
+	key, err := kcpcache.DeletionHandlingMetaClusterNamespaceKeyFunc(obj)
```

Comment:
As per the previous comment. This is only used for enqueuing.

Reply:
are partition names unique across workspace?

Reply:
I don't think so, does it mean we always have to find a local partition for the given slice?

Reply:
would it make sense to list endpoints for the given `cluster` and check which ones reference a particular name instead?

Reply:
> does it mean we always have to find a local partition for the given slice?

yes that is the intended design, partition and APIExportEndpointSlices being in the same workspace. There can however be multiple of them in different workspaces for covering different regions for instance.
Otherwise we would need to make them global and at this point in time I don't see the need for it.

> would it make sense to list endpoints for the given cluster and check which ones reference a particular name instead?

there is max one endpoint per shard, the Partition resource allows to specify shards for which endpoints get added to the APIExportEndpointSlice.